src/PublicApiAnalyzers/Core/CodeFixes/NullableEnablePublicApiFix.cs

```diff
@@ -80,16 +81,21 @@ protected override async Task<Solution> GetChangedSolutionAsync(CancellationToke
             {
                 var updatedPublicSurfaceAreaText = new List<KeyValuePair<DocumentId, SourceText>>();
 
-                foreach (KeyValuePair<Project, ImmutableArray<Diagnostic>> pair in _diagnosticsToFix)
+                using var uniqueShippedDocuments = PooledHashSet<string>.GetInstance();
+                foreach (var project in _projectsToFix)
                 {
-                    Project project = pair.Key;
                     TextDocument? shippedDocument = DeclarePublicApiFix.GetShippedDocument(project);
-                    SourceText? shippedSourceText = shippedDocument is null ? null : await shippedDocument.GetTextAsync(cancellationToken).ConfigureAwait(false);
-                    if (shippedSourceText is object)
+                    if (shippedDocument == null ||
+                        shippedDocument.FilePath != null && !uniqueShippedDocuments.Add(shippedDocument.FilePath))
```

Comment:
I would add parenthesis around the and block just to avoid confusion for maintenance later on. Feel free to ignore.

```suggestion
                        (shippedDocument.FilePath != null && !uniqueShippedDocuments.Add(shippedDocument.FilePath)))
```

Reply:
This is common code style in both Roslyn reps and we only put parentheses where either both sides of or have complex expressions or parentheses are needed for precedence.

Reply:
Fine by me :)